docs/DeveloperGuide.md

```diff
@@ -634,6 +634,155 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
       Use case ends.
 
+**Use case: UC17 - Add a lesson**
+
+**MSS**
+
+1. User requests to add a lesson.
+2. User provides the parameters of the lesson.
+3. Tutor's Pet adds the lesson.
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. The parameters provided are invalid.
+
+    * 2a1. Tutor's Pet shows an error message.
+
+      Use case resumes at step 1.
+
+* 3a. The lesson already exists.
+
+    * 3a1. Tutor's Pet shows an error message.
+
+      Use case resumes at step 1.
+
+**Use case: UC18 - Edit a lesson**
+
+**MSS**
+
+1.  User requests to edit a specific lesson.
+2.  User provides the parameters that are to be edited.
+3.  Tutor's Pet edits the lesson's information.
+
+    Use case ends.
+
+**Extensions**
+
+* 1a. The given index is invalid.
+
+    * 1a1. Tutor's Pet shows an error message.
+
+      Use case resumes at step 1.
+
+* 2a. The parameters provided are invalid.
+
+    * 2a1. Tutor's Pet shows an error message.
+
+      Use case resumes at step 2.
```

Comment:
should this use case resume at step 1 instead? since user will need to request to edit a specific lesson again..